lib/screen/bloc/accounts/accounts_bloc.dart

```diff
@@ -435,22 +268,42 @@ class AccountsBloc extends AuBloc<AccountsEvent, AccountsState> {
     return existingConnections.first;
   }
 
-  Future<Account?> getAccountPersona(Persona persona) async {
-    final nameFuture = persona.wallet().getName();
-    final ethAddress = await persona.wallet().getETHEip55Address();
-    if (ethAddress.isEmpty) return null;
-    var name = await nameFuture;
-
-    if (name.isEmpty) {
-      name = persona.name;
+  Future<List<Account>> getAccountPersona(List<WalletAddress> walletAddresses,
+      {bool addDefault = false}) async {
+    final personas = await _cloudDB.personaDao.getPersonas();
+    final List<WalletAddress> addresses = [];
+    addresses.addAll(walletAddresses);
+    if (walletAddresses.isEmpty && addDefault) {
+      final defaultPersona =
+          personas.firstWhere((element) => element.isDefault());
+      final defaultAddresses =
+          await defaultPersona.insertAddress(WalletType.Autonomy);
+      addresses.addAll(defaultAddresses);
+    }
+    List<Account> accounts = [];
+    for (var e in addresses) {
+      final persona =
+          personas.firstWhereOrNull((element) => element.uuid == e.uuid);
+      if (persona != null) {
+        accounts.add(Account(
+            key: e.address,
+            persona: persona,
+            name: e.name ?? e.cryptoType,
+            blockchain: e.cryptoType,
+            walletAddress: e,
+            accountNumber: e.address,
+            createdAt: e.createdAt));
+      }
     }
+    return accounts;
+  }
 
-    final account = Account(
-        key: persona.uuid,
-        persona: persona,
-        name: name,
-        accountNumber: ethAddress,
-        createdAt: persona.createdAt);
-    return account;
+  int _compareAccount(Account a, Account b) {
+    final aDefault = a.persona?.defaultAccount ?? 0;
+    final bDefault = b.persona?.defaultAccount ?? 0;
+    if (aDefault != bDefault) {
+      return bDefault.compareTo(aDefault);
+    }
+    return a.createdAt.compareTo(b.createdAt);
```

Comment:
Can we use SQL to sort here?

Reply:
accounts are not necessarily from 1 table, so even if we sort in SQL query, we still have to sort again